pkg/ingester/ingester.go

```diff
@@ -3764,6 +3758,38 @@ func (i *Ingester) ShutdownHandler(w http.ResponseWriter, _ *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
+// startReadRequest tries to start a read request.
+// If it was successful, startReadRequest returns a function that should be
+// called to finish the started read request once the request is completed.
+// If it wasn't successful, the causing error is returned. In this case no
+// function is returned.
+func (i *Ingester) startReadRequest() (func(error), error) {
+	start := time.Now()
+	acquiredCircuitBreakerPermit, err := i.circuitBreaker.tryAcquirePermit()
+	if err != nil {
+		return nil, err
+	}
+	err = i.checkAvailableForRead()
+	if err != nil {
+		return nil, err
+	}
+	err = i.checkReadOverloaded()
+	if err != nil {
+		return nil, err
+	}
+	return func(err error) {
+		i.finishReadRequest(start, acquiredCircuitBreakerPermit, err)
+	}, nil
+}
+
+// finishReadRequest finishes a previously started read request, whose state is described
+// by the passed readRequestState object.
+func (i *Ingester) finishReadRequest(startTime time.Time, acquiredCircuitBreakerPermit bool, err error) {
+	if acquiredCircuitBreakerPermit {
+		i.circuitBreaker.finishReadRequest(time.Since(startTime), err)
+	}
+}
```

Comment:
IMO the `i.finishReadRequest` doesn't add any value now.

```suggestion
	return func(err error) {	
		if acquiredCircuitBreakerPermit {
			i.circuitBreaker.finishReadRequest(time.Since(startTime), err)
		}
	}, nil
}
```

Reply:
I was using it in a test, but I will remove it.